Tasks/AzureRmWebAppDeployment/operations/AzureAppServiceUtility.ts

```diff
@@ -0,0 +1,228 @@
+            return new Kudu(publishingCredentials.properties["scmUri"], publishingCredentials.properties["publishingUserName"], publishingCredentials.properties["publishingPassword"]);
+        }
+
+        throw Error(tl.loc('KuduSCMDetailsAreEmpty'));
+    }
+
+    public async getPhysicalPath(virtualApplication: string): Promise<string> {
+
+        if(!virtualApplication) {
+            return '/site/wwwroot';
+        }
+
+        virtualApplication = (virtualApplication.startsWith("/")) ? virtualApplication.substr(1) : virtualApplication;
+
+        var appConfigSettings = await this._appService.getConfiguration();
+        var virtualApplicationMappings = appConfigSettings.properties.virtualApplications;
+        var phyicalToVirtualPathMap = this._getPhysicalToVirtualPathMap(virtualApplication, virtualApplicationMappings);
+
+        if(!phyicalToVirtualPathMap) {
+            throw Error(tl.loc("VirtualApplicationDoesNotExist", virtualApplication));
+        }
+
+        tl.debug(`Virtual Application Map: Physical path: '${phyicalToVirtualPathMap.physicalPath}'. Virtual path: '${phyicalToVirtualPathMap.virtualPath}'.`);
+        return phyicalToVirtualPathMap.physicalPath;
+    }
+
+    private _getPhysicalToVirtualPathMap(virtualApplication: string, virtualApplicationMappings: any): any {
+        // construct URL depending on virtualApplication or root of webapplication 
+        var physicalPath = null;
+        var virtualPath = "/" + virtualApplication;
+        
+        for( var mapping of virtualApplicationMappings ) {
+            if(mapping.virtualPath.toLowerCase() == virtualPath.toLowerCase()) {
+                physicalPath = mapping.physicalPath;
+                break;
+            }
+        }
+        return physicalPath ? {
+            'virtualPath': virtualPath,
+            'physicalPath': physicalPath
+        }: null;
+    }
+
+    public async updateConfigurationSettings(properties: any) : Promise<void> {
+        for(var property in properties) {
+            if(properties[property].value !== undefined) {
+                properties[property] = properties[property].value;
+            }
+        }
+
+        console.log(tl.loc('UpdatingAppServiceConfigurationSettings', JSON.stringify(properties)));
+        await this._appService.patchConfiguration({'properties': properties});
+        console.log(tl.loc('UpdatedAppServiceConfigurationSettings'));
+    }
+
+    public async updateAndMonitorAppSettings(properties: any): Promise<void> {
+        for(var property in properties) {
+            if(properties[property].value !== undefined) {
+                properties[property] = properties[property].value;
+            }
+        }
+        
+        console.log(tl.loc('UpdatingAppServiceApplicationSettings', JSON.stringify(properties)));
+        await this._appService.patchApplicationSettings(properties);
+        var kuduService = await this.getKuduService();
+        var interator: number = 6;
```

Comment:
Nit. typo in iterator spelling.

How did you arrive at 6?

Reply:
Retrying for 1 minute with 10 seconds interval.